setup.py

```diff
@@ -1,41 +1,55 @@
 #! /usr/bin/env python
 import setuptools  # noqa; we are using a setuptools namespace
-from numpy.distutils.core import setup
-import pyglmnet
+import os
+
+
+def get_version():
+    """
+    Get the version without importing, so as not to invoke dependency
+    requirements.
+    """
+    base, _ = os.path.split(os.path.realpath(__file__))
+    file = os.path.join(base, "pyglmnet", "__init__.py")
+
+    for line in open(file, "r"):
+        if "__version__" in line:
+            return line.split("=")[1].strip().strip("'").strip('"')
+
 
 descr = """Elastic-net regularized generalized linear models."""
 
-DISTNAME = 'pyglmnet'
+DISTNAME = "pyglmnet"
 DESCRIPTION = descr
-MAINTAINER = 'Pavan Ramkumar'
-MAINTAINER_EMAIL = 'pavan.ramkumar@gmail.com'
-LICENSE = 'MIT'
-DOWNLOAD_URL = 'https://github.com/glm-tools/pyglmnet.git'
-VERSION = pyglmnet.__version__
+MAINTAINER = "Pavan Ramkumar"
+MAINTAINER_EMAIL = "pavan.ramkumar@gmail.com"
+LICENSE = "MIT"
+DOWNLOAD_URL = "https://github.com/glm-tools/pyglmnet.git"
+VERSION = get_version()  # Get version without importing
+INSTALL_REQUIRES = ["numpy>=1.11", "scipy>=0.17", "scikit-learn>=0.18", "pytest"]
 
 if __name__ == "__main__":
-    setup(name=DISTNAME,
-          maintainer=MAINTAINER,
-          maintainer_email=MAINTAINER_EMAIL,
-          description=DESCRIPTION,
-          license=LICENSE,
-          version=VERSION,
-          download_url=DOWNLOAD_URL,
-          long_description=open('README.rst').read(),
-          classifiers=[
-              'Intended Audience :: Science/Research',
-              'Intended Audience :: Developers',
-              'License :: OSI Approved',
-              'Programming Language :: Python',
-              'Topic :: Software Development',
-              'Topic :: Scientific/Engineering',
-              'Operating System :: Microsoft :: Windows',
-              'Operating System :: POSIX',
-              'Operating System :: Unix',
-              'Operating System :: MacOS',
-          ],
-          platforms='any',
-          packages=[
-              'pyglmnet'
-          ],
-          )
+    setuptools.setup(
+        name=DISTNAME,
+        maintainer=MAINTAINER,
+        maintainer_email=MAINTAINER_EMAIL,
+        description=DESCRIPTION,
+        license=LICENSE,
+        version=VERSION,
+        download_url=DOWNLOAD_URL,
+        install_requires=INSTALL_REQUIRES,
```

Comment:
I prefer people to install the requirements using requirements.txt explicitly rather than installing them automatically this way.

Reply:
Any particular reason why? My understanding is that `install_requires` should specify what the project [minimally needs to run correctly](https://packaging.python.org/discussions/install-requires-vs-requirements/). Scipy, sklearn, pandas etc all seem to use `install_requires` for this purpose.

Reply:
Right, two reasons. First, not all dependencies are hard dependencies. For example, pandas is required only for manipulating one dataset. So every user should not be forced to install it. Second, it messes up your existing environment without your knowledge. I prefer to install numpy/scipy etc using conda rather than pip

Reply:
I see. Regarding the first point: `install_requires` would only contain hard dependencies, not `pytest`, `scikit-learn`, `flake8`, etc. Regarding the second point: I too prefer to install from conda rather than pip, and I see your point.

To play devils advocate though: a user without knowledge of environments, who in addition does not have numpy/scipy installed, would probably prefer everything to be handled automatically. A user with knowledge of environments can just delete it and start afresh anyway if he screws it up or regrets letting pip install numpy/scipy.

Anyway, I see your point and have removed it.